bin.js

```diff
@@ -60,6 +60,26 @@ if (argv[0] == 'server') {
   config.keys = keys
   var server = createSbot(config)
 
+  server.on('rpc:connect', function (rpc, isClient) {
+    if (rpc.id == '@' + keys.public) return
+
+    pull(
+      server.friends.hopStream(),
+      pull.drain(s => {
+        try {
+          var hops = s[rpc.id]
+          if (hops == undefined || hops > config.friends.hops) {
+            server.emit('log:info', ['SBOT', 'closing connection outside hops: ', hops])
+            rpc.close(true, function() { })
+          }
+        } catch (e) {
```

Comment:
Wow, this is really cool. I appreciate you taking this on. I have a couple of questions, just for my better understanding:

- Which sorts of errors can we expect to happen within this `try {}` block?
- Would `hops === undefined` have the same effect here? Or `hops == null`?
- Can `rpc.close()` be called with only one argument? I'm not sure if [this](https://github.com/ssbc/scuttlebot/blob/f6e2d256794d075ec47526344d41a008b5a814bd/index.js#L61) is the right code, but it looks like the argument could be optional.